modules/post/linux/gather/enum_commands.rb

```diff
@@ -9,77 +9,89 @@ class MetasploitModule < Msf::Post
 
   def initialize
     super(
-      'Name'         => 'Testing commands needed in a function',
-      'Description'  => %q{
-        This module will be applied on a session connected to a shell. It will check which commands are available in the system.
+      'Name' => 'Gather Available Shell Commands',
+      'Description' => %q{
+        This module will check which shell commands are available on a system."
       },
-      'Author'       => 'Alberto Rafael Rodriguez Iglesias <albertocysec[at]gmail.com>',
-      'License'      => MSF_LICENSE,
-      'Platform'     => ['linux'],
-      'SessionTypes' => ['shell', 'meterpreter']
+      'Author' => 'Alberto Rafael Rodriguez Iglesias <albertocysec[at]gmail.com>',
+      'License' => MSF_LICENSE,
+      'Platform' => ['linux', 'unix'],
+      'SessionTypes' => ['shell', 'meterpreter'],
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'Reliability' => [],
+        'SideEffects' => []
+      }
     )
-    register_options(
-      [
-        OptString.new('DIR', [false, 'Optional directory name to list, default current session path',''])
-      ])
+    register_options([
+      OptString.new('DIR', [false, 'Optional directory name to list (in addition to default system PATH and common paths)', ''])
+    ])
   end
 
   DIRS = [
-    "/root/local/bin/",
-    "/usr/local/sbin/",
-    "/usr/local/bin/",
-    "/usr/sbin/",
-    "/usr/bin/",
-    "/sbin/",
-    "/bin/",
-    "/usr/local/go/bin/"
-  ]
+    '/root/local/bin',
+    '/usr/local/sbin',
+    '/usr/local/bin',
+    '/usr/sbin',
+    '/usr/bin',
+    '/sbin',
+    '/bin',
+    '/usr/local/go/bin'
+  ].freeze
 
   def run
-    dir = datastore['DIR']
+    path = get_path
+
+    print_warning('System PATH is empty!') if path.blank?
+
+    paths = []
+    path.split(':').each do |p|
+      paths << p.chomp('/')
+    end
+
+    common_dirs = DIRS
```

Comment:
```suggestion
    common_dirs = DIRS.dup
```

Should fix the error:

```
msf6 post(linux/gather/enum_commands) > run session=-1 dir=/usr/bin

[-] Post failed: FrozenError can't modify frozen Array: ["/root/local/bin", "/usr/local/sbin", "/usr/local/bin", "/usr/sbin", "/usr/bin", "/sbin", "/bin", "/usr/local/go/bin"]
[-] Call stack:
[-]   /Users/adfoster/Documents/code/metasploit-framework/modules/post/linux/gather/enum_commands.rb:53:in `run'
[*] Post module execution completed
```

Reply:
Oops. I specifically recall wondering whether that would be a problem. Apparently I did nothing about it.

There's no real reason why this has to be a global constant. I'll move it into `exploit` method.

Reply:
Fixed